### PR TITLE
Add moon phase expression

### PR DESCRIPTION
**Pull request by UnderscoreTud**
### Description
This PR adds an expression to get the current moon phase of a world

---
**Target Minecraft Versions:** 1.16+
**Requirements:** Paper
**Related Issues:** https://github.com/SkriptLang/Skript/issues/746

### CODE REVIEW

File: src/main/java/ch/njol/skript/classes/data/BukkitClasses.java [hunk 1]
```diff
@@ -1836,5 +1837,34 @@ public String toVariableNameString(Attribute a) {
...
+
+		if (Skript.classExists("io.papermc.paper.world.MoonPhase")) {
+			EnumUtils<MoonPhase> moonPhases = new EnumUtils<>(MoonPhase.class, "moon phases");
+			Classes.registerClass(new ClassInfo<>(MoonPhase.class, "moonphase")
+				.user("(lunar|moon) ?phases?")
+				.name("Moon Phase")
+				.description("Represents the phase of a moon.")
+				.usage(moonPhases.getAllNames())
+				.since("INSERT VERSION")
+				.requiredPlugins("Paper")
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/classes/data/BukkitClasses.java:1849]:
```suggestion
				.requiredPlugins("Paper 1.16+")
```

File: src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Moon Phase")
+@Description("The current moon phase of a world")
+@Examples({
+	"if moon phase of player's world is full moon:",
+	"\tsend \"today's a full moon!\""
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java:39]:
```suggestion
@RequiredPlugins("Paper 1.16+")
```

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Nice PR 🎉

**Review by AyhamAl-Ali — APPROVED** [review]

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import io.papermc.paper.world.MoonPhase;
+import org.bukkit.World;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Moon Phase")
+@Description("The current moon phase of a world")
+@Examples({
+	"if moon phase of player's world is full moon:",
+	"\tsend \"today's a full moon!\""
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java:36]:
```suggestion
		"\tsend \"today's a full moon!\""
```

File: src/main/java/ch/njol/skript/classes/data/BukkitClasses.java [hunk 4]
```diff
@@ -1836,5 +1837,34 @@ public String toVariableNameString(Attribute a) {
...
+					}
+
+					@Override
+					public String toString(MoonPhase moonPhase, int flags) {
+						return moonPhases.toString(moonPhase, flags);
+					}
+
+					@Override
+					public String toVariableNameString(MoonPhase moonPhase) {
+						return toString(moonPhase, 0);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/classes/data/BukkitClasses.java:1864]:
```suggestion
						return moonPhase.name();
```
I think this is preferred? would be consistent with your enum class info pr too

File: src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import io.papermc.paper.world.MoonPhase;
+import org.bukkit.World;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Moon Phase")
+@Description("The current moon phase of a world")
+@Examples({
+	"if moon phase of player's world is full moon:",
+		"\tsend \"today's a full moon!\""
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java:36]:
```suggestion
		"\tsend \"Watch for the wolves!\""
```
maybe this is more fun 🤣

File: src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import io.papermc.paper.world.MoonPhase;
+import org.bukkit.World;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Moon Phase")
+@Description("The current moon phase of a world")
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprMoonPhase.java:33]:
```suggestion
@Description("The current moon phase of a world.")
```

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
Minor formatting - fun stuff!!

**Review by APickledWalrus — APPROVED** [review]

**Comment by TheLimeGlass** [comment]
@UnderscoreTud conflicts

**Comment by UnderscoreTud** [comment]
> @UnderscoreTud conflicts

alright fixed

File: src/main/java/ch/njol/skript/classes/data/BukkitClasses.java [hunk 7]
```diff
@@ -28,16 +28,10 @@
...
-import ch.njol.skript.classes.EnumClassInfo;
-import ch.njol.skript.classes.Parser;
-import ch.njol.skript.classes.Serializer;
+import ch.njol.skript.classes.*;
 import ch.njol.skript.lang.util.SimpleLiteral;
-import ch.njol.skript.util.BlockUtils;
-import ch.njol.skript.util.EnchantmentType;
-import ch.njol.skript.util.PotionEffectUtils;
-import ch.njol.skript.util.StringMode;
+import ch.njol.skript.util.*;
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/classes/data/BukkitClasses.java:33]: No wild cards

File: src/main/java/ch/njol/skript/classes/data/BukkitClasses.java [hunk 8]
```diff
@@ -28,16 +28,10 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
-import ch.njol.skript.classes.ClassInfo;
-import ch.njol.skript.classes.ConfigurationSerializer;
-import ch.njol.skript.classes.EnumClassInfo;
-import ch.njol.skript.classes.Parser;
-import ch.njol.skript.classes.Serializer;
+import ch.njol.skript.classes.*;
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/classes/data/BukkitClasses.java:31]: No wild cards

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

**Review by TheLimeGlass — APPROVED** [review]

**Pull request merged**